server/routes/api/comments/comments.ts

```diff
@@ -61,18 +62,53 @@ router.post(
   checkCommentingEnabled(),
   validate(T.CollectionsListSchema),
   async (ctx: APIContext<T.CollectionsListReq>) => {
-    const { sort, direction, documentId } = ctx.input.body;
+    const { sort, direction, documentId, collectionId } = ctx.input.body;
     const { user } = ctx.state.auth;
 
-    const document = await Document.findByPk(documentId, { userId: user.id });
-    authorize(user, "read", document);
-
-    const comments = await Comment.findAll({
-      where: { documentId },
-      order: [[sort, direction]],
-      offset: ctx.state.pagination.offset,
-      limit: ctx.state.pagination.limit,
-    });
+    let comments;
+    if (documentId) {
+      const document = await Document.findByPk(documentId, { userId: user.id });
+      authorize(user, "read", document);
+      comments = await Comment.findAll({
+        where: { documentId },
+        order: [[sort, direction]],
+        offset: ctx.state.pagination.offset,
+        limit: ctx.state.pagination.limit,
+      });
+    } else if (collectionId) {
+      const collection = await Collection.findByPk(collectionId);
+      authorize(user, "read", collection);
+      comments = await Comment.findAll({
+        include: [
+          {
+            model: Document,
+            where: {
```

Comment:
```suggestion
            where: {
              required: true,
```

Reply:
Also here, you've suggested the `required: true` to be inside the `where` clause. Is that the right place, or should it be outside `where`? Like mentioned [here](https://sequelize.org/docs/v6/advanced-association-concepts/eager-loading/#required-eager-loading)

Reply:
outside yes, apologies